src/main/java/analysis_use_case/Analysis.java

```diff
@@ -14,62 +18,73 @@ public Analysis(Inventory inventory){
-     * Check if each item is quantified by weight, if so print the information accordingly
-     * Print out the total revenue of the inventory
+     * Return a revenue breakdown
+     * Each index should contain the item's name, quantity sold, selling price, total revenue for that product
+     * Check if each item is quantified by weight, if so present the information accordingly
+     * The last index should have the total revenue of the inventory
      */
-    public void totalRevenueBreakdown(){
-        System.out.println("Revenue:");
-        for (InventoryItem item : inventory.getItems()){
-            if (item.getIsWeight()){
-                System.out.println(item.getName() + ": kg sold: " + item.getQuantitySold() +
-                                   " price per kg: " + item.getSellPrice() + " total revenue for product: " +
-                                   item.getQuantitySold() * item.getSellPrice());
+    public ArrayList<String> totalRevenueBreakdown() {
+        ArrayList<String> RevenueItems = new ArrayList<>();
+        RevenueItems.add("Revenue:");
+        for (InventoryItem item : inventory.getItems()) {
+            String ItemBreakdown;
+            if (item.getIsWeight()) {
+                ItemBreakdown = item.getName() + ": kg sold: " + item.getQuantitySold() +
+                        " price per kg: " + item.getSellPrice() + " total revenue for product: " +
+                        item.getQuantitySold() * item.getSellPrice();
 
             } else{
-                System.out.println(item.getName() + ": quantity sold: " + item.getQuantitySold() +
+                ItemBreakdown = item.getName() + ": quantity sold: " + item.getQuantitySold() +
                         " price per product: " + item.getSellPrice() + " total revenue for product: " +
-                        item.getQuantitySold() * item.getSellPrice());
+                        item.getQuantitySold() * item.getSellPrice();
             }
-        }
-        System.out.println("Total Revenue: " + this.getTotalRevenue());
+            RevenueItems.add(ItemBreakdown);
 
+        }
+        RevenueItems.add("Total Revenue: " + this.getTotalRevenue());
+        return RevenueItems;
     }
 
     /**
-     * Prints a cost breakdown line by line
-     * Each line should contain the item's name, quantity bought, buy price, total cost for that product
-     * Check if each item is quantified by weight, if so print the information accordingly
-     * Print out the total cost of the inventory
+     * Returns a cost breakdown
+     * Each index should contain the item's name, quantity bought, buy price, total cost for that product
+     * Check if each item is quantified by weight, if so present the information accordingly
+     * The last index should have the total cost of the inventory
      */
-    public void totalCostsBreakdown(){
-        System.out.println("Costs: ");
+    public ArrayList<String> totalCostsBreakdown(){
```

Comment:
Instead of printing the information (which was violating clean architecture) I store the information in a ArrayList of Strings where each index is a breakdown of the item's cost. This made it easier to implement the UI.